File: src/main/java/com/groupproject/softwaretutorials/project/Project.java
package com.groupproject.softwaretutorials.project;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import lombok.Data;
import java.io.Serializable;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;

@Entity
@Data
public class Project implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long projectId;

    
    @Column(length = 50)
    private String title;

    
    @Column(length = 1000)
    private String problemStatement;
}

File: front-end/lib/models/project.dart
class Project{
	int? projectId;
    String? title;
    String? problemStatement;
    Project({this.title, this.problemStatement, this.projectId});
	
	factory Project.fromJson(Map<String, dynamic?>? json){
		if (json == null){
			return Project();
		}
		return Project(
			projectId: json['projectId'],
			title: json['title'],
			problemStatement: json['problemStatement']
		);
	}
}

class ProjectSubmission {
    String? projectUrl;
	ProjectSubmission({this.projectUrl});
	
	factory ProjectSubmission.fromJson(Map<String, dynamic?> json){
	
	return ProjectSubmission(projectUrl: json['projectUrl']);
	}
	
}
